### PR TITLE
#465 added a test for forgot password.

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #465
 - Added a test for forgot the password logic.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#521](https://github.com/timelesslounge/timelessis/issues/521) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#521](https://github.com/timelesslounge/timelessis/issues/521) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/b47fdbe4-6e16-4b93-939e-a8646498611f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_auth_test.py [hunk 1]
```diff
@@ -63,12 +63,23 @@ def test_activate(client):
...
-     change it for the user with the received e-mail, find it changed.
-"""
-
+def test_forgot_password_routine(db_session, client):
+    employee = factories.EmployeeFactory(
+        username="uname",
+        password=auth_hash("pass"),
+        email="mail@mail.com"
+    )
+    db_session.add(employee)
```
Comment (vryazanov) [inline tests/integration/it_auth_test.py]: @ammaratef45 Factory already commited `employee`, so you dont need to use `db_session` directly.

File: tests/integration/it_auth_test.py [hunk 2]
```diff
@@ -63,12 +63,23 @@ def test_activate(client):
...
+    db_session.commit()
+    error = login("uname", "pass")
```
Comment (vryazanov) [inline tests/integration/it_auth_test.py]: @ammaratef45 lets do `error = login(employee.username, "pass")`

File: tests/integration/it_auth_test.py [hunk 3]
```diff
@@ -63,12 +63,23 @@ def test_activate(client):
...
+    assert not error
+    assert flask.session["user_id"] == employee.id
```
Comment (vryazanov) [inline tests/integration/it_auth_test.py]: @ammaratef45 do we really need these 2 lines? I think we already have tests for `login` function, so we do not need to double check that it works

**Comment by ammaratef45** [comment]
@vryazanov Thank you very much.
Please check

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 5 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#521](https://github.com/timelesslounge/timelessis/issues/521) is now out of scope

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#521](https://github.com/timelesslounge/timelessis/issues/521) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Pull request merged**